core/lib/dal/src/tokens_dal.rs

```diff
@@ -79,34 +76,25 @@ impl TokensDal<'_, '_> {
             .collect())
     }
 
+    /// Removes token records that were deployed after `block_number`.
     pub async fn rollback_tokens(&mut self, block_number: MiniblockNumber) -> sqlx::Result<()> {
+        let all_token_addresses = self.get_all_l2_token_addresses().await?;
+        let token_deployment_data = self
+            .storage
+            .storage_logs_dal()
+            .filter_deployed_contracts(all_token_addresses.into_iter(), None)
+            .await?;
+        let token_addresses_to_be_removed: Vec<_> = token_deployment_data
+            .into_iter()
+            .filter_map(|(address, deployed_at)| (deployed_at > block_number).then_some(address.0))
+            .collect();
```

Comment:
Wow I think it fixes https://linear.app/matterlabs/issue/PLA-726/block-reversion-tool-tokens-are-not-rolled-back-properly ?? That's great!